pkg/services/cloudmigration/cloudmigrationimpl/cloudmigration.go

```diff
@@ -777,3 +788,33 @@ func (s *Service) getLocalEventId(ctx context.Context) (string, error) {
 
 	return anonId, nil
 }
+
+// getResourcesWithPluginWarnings iterates through each resource and, if a non-core datasource, applies a warning that we only support core
+func (s *Service) getResourcesWithPluginWarnings(ctx context.Context, results []cloudmigration.CloudMigrationResource) ([]cloudmigration.CloudMigrationResource, error) {
+	reqCtx := contexthandler.FromContext(ctx)
+	for i := 0; i < len(results); i++ {
+		r := results[i]
+
+		if r.Type == cloudmigration.DatasourceDataType &&
+			r.Error == "" { // any error returned by GMS takes priority
+			ds, err := s.dsService.GetDataSource(ctx, &datasources.GetDataSourceQuery{
```

Comment:
Is this hitting the database `n` times? Ideally, the results could be fetched with a single query

Reply:
@PoorlyDefinedBehaviour  I can change this to call `GetAllDatasources` and do a lookup instead.